.github/workflows/testsuite.yml

```diff
@@ -464,3 +464,53 @@ jobs:
       - name: Run Tests
         run: |
           make minitest_notty
+
+  #     _    ____    _    _   _
+  #    / \  / ___|  / \  | \ | |
+  #   / _ \ \___ \ / _ \ |  \| |
+  #  / ___ \ ___) / ___ \| |\  |
+  # /_/   \_\____/_/   \_\_| \_|
+
+  # Life is pain, highness. Anyone who says differently is selling something.
+
+  ASAN:
+    name: "ASAN"
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    needs: sanity_check
+    if: needs.sanity_check.outputs.run_all_jobs == 'true'
+
+    env:
+      PERL_SKIP_TTY_TEST: 1
+      CONTINUOUS_INTEGRATION: 1
+      WORKSPACE: ${{ github.workspace }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # exercise a variety of build options
+        # threads often cause build issues
+        CONFIGURE_ARGS:
+          - "-Dusethreads"
```

Comment:
Do we want/need also with and without debugging for this ASAN build? I've been noticing recently that gcc can act quite differently (and sometimes buggily) in the presence of asserts. I've no clue what the cost/benefit ratio looks like though.

Reply:
> Do we want/need also with and without debugging for this ASAN build? I've been noticing recently that gcc can act quite differently (and sometimes buggily) in the presence of asserts. I've no clue what the cost/benefit ratio looks like though.

I don't either. In that, I believe that everything runs in parallel, and we don't seem to saturate the available hardware, meaning that "however many" Linux tests we add, cygwin will still finish last. Between the "ASAN" and "PERL_UNICODE" there are two jobs with `-Dusethreads` and only one with (implied) `-Uusethreads` so I guess moving the precise location of the `-DDEBUGGING` and having one (more) job here with (`no threads` and `no DEBUGGING`) would cover this case.

@rjbs - do you know what accounting policy github has for workflow jobs, and if we're at risk of hitting it and hence having useful jobs denied?

Reply:
> @rjbs - do you know what accounting policy github has for workflow jobs, and if we're at risk of hitting it and hence having useful jobs denied?

I recall we tripped up on that not so long ago, but I don't remember the details; I think it may have been @atoomic that fixed things that time.

Reply:
> > @rjbs - do you know what accounting policy github has for workflow jobs, and if we're at risk of hitting it and hence having useful jobs denied?
> 
> I recall we tripped up on that not so long ago, but I don't remember the details; I think it may have been @atoomic that fixed things that time.

I can't find *much* documentation that gives me confidence in an answer.


The only thing that mentions limits (well, free quotas) is https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions

I assume that we are "GitHub Free for organizations" and hence that gives us 2,000 minutes per month.

Which works out at 64 minutes daily. Which I think we'd burn through on one push request.

Which must mean that this applies:

> GitHub Actions usage is free for both public repositories and self-hosted runners.

Which seems to imply that it's unlimited. Or more realistically, "unlimited".

Sigh. So I still don't *really* know.

Reply:
> > GitHub Actions usage is free for both public repositories and self-hosted runners.
> 
> Which seems to imply that it's unlimited. Or more realistically, "unlimited".

Ah, my recollection of tripping up on this before must have been for the security repo, which is _not_ a public repository. I think we had the mirroring set up such that every CI run for the public repo was duplicated; and that we eventually worked out what was happening and changed the set up so that the duplication no longer occurs.

So I think you're right, that we are "unlimited".